Patch-Source: https://github.com/ryanoasis/nerd-fonts/pull/764 (updated)
--
From 9b3caeab1bb7fd173ba41bed5fc5b6dd1664ebb4 Mon Sep 17 00:00:00 2001
From: Fini Jastrow <ulf.fini.jastrow@desy.de>
Date: Thu, 13 Jan 2022 17:43:53 +0100
Subject: [PATCH] Revert "Remove negative bearings on 2048-em glyphs"

This reverts commit 59c45ba4eff539d2a83a7d7004f056b9860d3253.

[why]
The commit breaks the non-mono Nerd Fonts for a lot of people.
The issue was not very good documented and investigation can not be
seen.

For the TITLE it should have affected only the 2048-em Symbols only
font, but in fact all patched fonts were changed. Maybe that was
intended, maybe not.

[how]
This will make the advance width again equal for all glyphs.
This enables the use of the non-mono variant in more terminals.
For wider symbols a space is now (again) needed after the symbol.
That is expected by a lot applications.

[note]
See Pull Request 764 for more details.

Signed-off-by: Fini Jastrow <ulf.fini.jastrow@desy.de>
---
 font-patcher | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/font-patcher b/font-patcher
index 480bf3ba1..1c5c96932 100755
--- a/font-patcher
+++ b/font-patcher
@@ -840,15 +840,16 @@
             align_matrix = psMat.translate(x_align_distance, y_align_distance)
             self.sourceFont[currentSourceFontGlyph].transform(align_matrix)
 
+            # Ensure after horizontal adjustments and centering that the glyph
+            # does not overlap the bearings (edges)
+            self.remove_glyph_neg_bearings(self.sourceFont[currentSourceFontGlyph])
+
             # Needed for setting 'advance width' on each glyph so they do not overlap,
             # also ensures the font is considered monospaced on Windows by setting the
             # same width for all character glyphs. This needs to be done for all glyphs,
             # even the ones that are empty and didn't go through the scaling operations.
+            # it should come after setting the glyph bearings
             self.set_glyph_width_mono(self.sourceFont[currentSourceFontGlyph])
-
-            # Ensure after horizontal adjustments and centering that the glyph
-            # does not overlap the bearings (edges)
-            self.remove_glyph_neg_bearings(self.sourceFont[currentSourceFontGlyph])
 
             # Check if the inserted glyph is scaled correctly for monospace
             if self.args.single:
